spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcTagKey.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2014-2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spectator.ipc;
+
+/**
+ * Standard dimensions used for IPC metrics.
+ */
+public enum IpcTagKey {
+
+  /** The library that produced the metric. */
+  owner("owner"),
+
+  /**
+   * Indicates whether or not the request was successful. See {@link IpcResult}
+   * for permitted values.
+   */
+  result("ipc.result"),
+
+  /**
+   * Dimension indicating a high level cause for the request failure. These groups
+   * are the same for all implementations to make it easier to query across all
+   * services and client implementations. See {@link IpcErrorGroup} for permitted
+   * values.
+   */
+  errorGroup("ipc.error.group"),
+
+  /**
+   * Implementation specific error code.
+   */
+  errorReason("ipc.error.reason"),
+
+  /**
+   * Indicates the attempt number for the request.
+   */
+  attempt("ipc.attempt"),
+
+  /**
+   * Indicates if this is the final attempt for the request. For example, if the client
+   * is configured to allow 1 retry, then the second attempt would be final. Acceptable
+   * values are {@code true} and {@code false}.
+   */
+  attemptFinal("ipc.attempt.final"),
+
+  /**
+   * Identifier for the client instance making the request. This is typically the name used
+   * to see the associated configuration settings for the client.
+   */
+  id("id"),
+
+  /**
+   * Eureka VIP used to select the server. If there are multiple vips in the list, then it
+   * should be the first vip that caused the server to get selected.
+   */
+  vip("ipc.vip"),
+
+  /**
+   * The server side endpoint that is being requested. This should be a bounded set that
+   * will have a small cardinality. <b>Do not pass in the unsanitized path value for a
+   * URI.</b>
+   */
+  endpoint("ipc.endpoint"),
+
+  /**
+   * Protocol used for the request. For example {@code http_1} or {@code http_2}.
+   */
+  protocol("ipc.protocol"),
+
+  /**
+   * Region where the client is located.
+   */
+  clientRegion("ipc.client.region"),
```

Comment:
I hadn't included ipc.client.region in the spec because I was thinking it was only really applicable for Zuul (as it's one of the few apps that send cross-region traffic). But maybe it makes sense to include it, even though it will generally be "none" (or same as nf.region?) for most apps?